splinterd/src/daemon.rs

```diff
@@ -1272,9 +1283,12 @@ impl SplinterDaemonBuilder {
             CreateError::MissingRequiredField("Missing field: heartbeat".to_string())
         })?;
 
-        let node_id = self.node_id.ok_or_else(|| {
-            CreateError::MissingRequiredField("Missing field: node_id".to_string())
-        })?;
+        let node_id = self
```

Comment:
This should remain an option here and not be generated if None, that happens in your get_node_id() function that is called in SplinterDaemon.start(). Not this means that it should not be returning an error here.